ros2launch/ros2launch/api/api.py

```diff
@@ -137,8 +139,14 @@ def parse_launch_arguments(launch_arguments: List[Text]) -> List[Tuple[Text, Tex
 def launch_a_python_launch_file(*, python_launch_file_path, launch_file_arguments, debug=False):
     """Launch a given Python launch file (by path) and pass it the given launch file arguments."""
     launch_service = launch.LaunchService(argv=launch_file_arguments, debug=debug)
+    context = rclpy.context.Context()
+    rclpy.init(args=launch_file_arguments, context=context)
```

Comment:
@wjwwood I'm afraid I'm not up on what kind of arguments we're expecting to pass to the context here so I'm not able to test this very well.  It looks like the launch_file_arguments get passed all the way through to become LaunchContext.argv which gets passed to rclpy.init up in ROSSpecificLaunchStartup so this should do the same thing, but I'm afraid I haven't checked

Reply:
I think this is right.